frontend/src/views/CalculateNutrients/CalculateNutrientsComponents/FertilizerModal.tsx

```diff
@@ -179,29 +240,98 @@ export default function FertilizerModal({
         setFertilizerUnits(data);
       }
     });
+
+    apiCache.callEndpoint('api/densityunits/').then((response) => {
+      if (response.status === 200) {
+        const { data } = response;
+        setDensityUnits(data);
+      }
+    });
+
+    apiCache.callEndpoint('api/liquidfertilizerdensities/').then((response) => {
+      if (response.status === 200) {
+        const { data } = response;
+        setLiqDensityFactors(data);
+      }
+    });
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const handleModalCalculate = (e: FormEvent) => {
-    e.preventDefault();
+  const isCustomFertilizerValid = () => {
+    if (formCustomFertilizer.nitrogen || formCustomFertilizer.nitrogen === 0) return false;
```

Comment:
I think you mean to ! this boolean condition.

Reply:
Absolutely.